src/api/console/buffer.rs

```diff
@@ -0,0 +1,28 @@
+use crate::types::Result;
+use std::io::Write;
+
+pub struct Buffer {
+    buf: Vec<u8>,
+}
+
+impl Buffer {
+    pub fn new() -> Self {
+        Self { buf: Vec::new() }
+    }
+
+    pub fn to_string(&self) -> Result<String> {
+        Ok(String::from_utf8(self.buf.clone())?)
+    }
+}
+
+impl Write for Buffer {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.buf.extend(buf);
+        Ok(buf.len())
```

Comment:
Did you want to fix this return value before merging?  If not, perhaps add a `// TODO: return correct return value`.